File: task-bazel/Facade/cpp-source/facades/Facade.cpp
//
// Created by Pavel Akhtyamov on 2019-03-20.
//

#include "Facade.h"

std::string Facade::CreateNormalGame() {
  return isaac_facade.CreateGame("normal");
}

std::string Facade::CreateHardGame() {
  return isaac_facade.CreateGame("hard");
}
std::string Facade::CreateVegiburger() {
  return burger_facade.createVegiburger();
}

File: JavaSource/src/main/java/ru/akhcheck/patterns/facade/facades/Facade.java
package ru.akhcheck.patterns.facade.facades;

/**
 * Фасад над двумя другими фасадами.
 *
 * <p>Показывает, что фасады складываются: каждый следующий уровень убирает ещё
 * часть подробностей. Наружу остаются три действия, за которыми стоят
 * строители, повар, фабрики комнат и сами комнаты.
 */
public class Facade {
    private final IsaacFacade isaacFacade = new IsaacFacade();
    private final BurgerFacade burgerFacade = new BurgerFacade();

    public String createNormalGame() {
        return isaacFacade.createGame("normal");
    }

    public String createHardGame() {
        return isaacFacade.createGame("hard");
    }

    public String createVegiburger() {
        return burgerFacade.createVegiburger();
    }
}
